### PR TITLE
feat: `insertcols!` receives kwarg `after`

### LINKED ISSUE #2613: Add `after` keyword argument to `insertcols!`
Currently you have to do this if you want to insert columns after a certain column:

```julia
df = DataFrame(a=1:2, b=3:4, c=5:6)
i = findfirst(==(:b), propertynames(df))
insertcols!(df, i+1, :x => 7:8)
```

It would be nice if there were a boolean keyword argument `after` which defaults to `false`.

### PR DESCRIPTION
**Pull request by pstorozenko**
Closes #2613

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -18,6 +18,9 @@
   (notably `PooledArray` and `CategoricalArray`) or when they contained only
   integers in a small range.
   ([#2812](https://github.com/JuliaData/DataFrames.jl/pull/2812))
+* the `insertcols!` function receives new keyword argument `after`
+  (with `false` default) that specifies if columns should be inserted after
+  or before `col`.
```
Comment (bkamins) [inline NEWS.md:23]:
```suggestion
  or before `col`.
  ([#2829](https://github.com/JuliaData/DataFrames.jl/pull/2829))
```

File: src/dataframe/dataframe.jl [hunk 2]
```diff
@@ -894,20 +917,21 @@ function insertcols!(df::DataFrame, col::ColumnIndex, name_cols::Pair{Symbol, <:
...
 
 insertcols!(df::DataFrame, name_cols::Pair{<:AbstractString, <:Any}...;
-            makeunique::Bool=false, copycols::Bool=true) =
+            after::Bool=false, makeunique::Bool=false, copycols::Bool=true) =
     insertcols!(df, (Symbol(n) => v for (n, v) in name_cols)...,
-                makeunique=makeunique, copycols=copycols)
+                after=after, makeunique=makeunique, copycols=copycols)
 
-function insertcols!(df::DataFrame, col::Int=ncol(df)+1; makeunique::Bool=false, name_cols...)
+function insertcols!(df::DataFrame, col::Int=ncol(df)+1; after::Bool=false, makeunique::Bool=false, name_cols...)
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:934]: this is not needed - this method was kept only to throw an error for a deprecated functionality.

Comment (bkamins) [inline src/dataframe/dataframe.jl:934]:
```suggestion
function insertcols!(df::DataFrame, col::Int=ncol(df)+1; makeunique::Bool=false, name_cols...)
```

File: test/dataframe.jl [hunk 3]
```diff
@@ -349,6 +349,32 @@ end
     @test_throws ArgumentError insertcols!(df, 2, y=2:3)
 end
 
+@testset "insertcols! after" begin
+    df = DataFrame(a=1:3)
+    insertcols!(df, :a, "b"=>2:4, after=true)
```
Comment (bkamins) [inline test/dataframe.jl:354]: add spaces around `=>` everywhere plese.

**Review by bkamins — APPROVED** [review]
Looks good. Let us wait a few days for other people to comment.

**Comment by bkamins** [comment]
@pstorozenko - this requires a simple merge conflict fix.

@nalimilan - could you please review the PR? Thank you!

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you. I will merge it once the CI passes.

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**